packages/nodes-base/nodes/GraphQL/GraphQL.node.ts

```diff
@@ -484,7 +487,20 @@ export class GraphQL implements INodeType {
 				} else {
 					if (typeof response === 'string') {
 						try {
-							response = JSON.parse(response);
+							if (typeof response === 'string') {
+								response = JSON.parse(response) as IDataObject;
+							}
+
+							// Check for errors in the response
+							if (response.errors && Array.isArray(response.errors)) {
+								// If the request format is 'graphql', throw an error suggesting to try JSON
+								if (requestFormat === 'graphql') {
```

Comment:
```suggestion
							if (response.errors && Array.isArray(response.errors) && requestFormat === 'graphql') {
```
Just some suggestions to make the code a bit more succinct :) I also don't think the comments are necessary for now since the code, in my opinion, is clear enough to indicate we're checking for errors and giving a suggestion in the error